modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultGenerator.java

```diff
@@ -533,9 +533,9 @@ public int compare(CodegenOperation one, CodegenOperation another) {
                     }
                 });
                 Map<String, Object> operation = processOperations(config, tag, ops, allModels);
-
+                URL url = URLPathUtils.getServerURL(openAPI);
                 operation.put("basePath", basePath);
-                operation.put("basePathWithoutHost", basePathWithoutHost);
+                operation.put("basePathWithoutHost", config.encodePath(url.getPath()).replaceAll("/$", ""));
```

Comment:
@ybelenko why do we need to replace `/$` with empty string?

Reply:
Thanks for the explanation

Reply:
@wing328 Maybe I can change `basePathWithoutHost` somewhere else? In some post process functions in Slim generator? Or maybe we should add `encodedBasepathWithoutHost` property? I think that this problem with non-latin characters exists in other generators as well.

Reply:
>  I think that this problem with non-latin characters exists in other generators as well.

I think so but I didn't have time to test and confirm.

Reply:
My take is to start with your fix for Slim generator and later we port this fix to the default codegen or other generators if users report similar issues.

Reply:
Yeah, I agree, it's not so important now. 

Important part is `basePathWithoutHost` value. Whether we should override it or put it in separated prop. Do you know how to override it inside `PhpSlimServerCodegen` codegen class?

What `contextPath` property stands for?